lib/Db/ItemMapper.php

```diff
@@ -492,29 +496,74 @@ public function readItem($itemId, $isRead, $lastModified, $userId)
         }
     }
 
-    /**
-     * NO-OP
-     *
-     * @param string $userId
-     *
-     * @return array
-     */
-    public function findAllFromUser(string $userId): array
+    public function update(Entity $entity): Entity
     {
-        return [];
+        $entity->setLastModified($this->time->getMicroTime());
+        return parent::update($entity);
     }
 
-    public function findFromUser(string $userId, int $id): Entity
+    public function insert(Entity $entity): Entity
     {
-        return $this->find($id, $userId);
+        $entity->setLastModified($this->time->getMicroTime());
+        return parent::insert($entity);
     }
 
     /**
-     * NO-OP
-     * @return array
+     * Remove deleted items.
+     *
+     * @return void
+     */
+    public function purgeDeleted(): void
+    {
+        $builder = $this->db->getQueryBuilder();
+        $builder->delete($this->tableName)
+            ->where('deleted_at != 0')
+            ->execute();
+    }
+    /**
+     * Performs a SELECT query with all arguments appened to the WHERE clause
+     * The SELECT will be performed on the current table and take the entity
+     * that is related for transforming the properties into column names
+     *
+     * Important: This method does not filter marked as deleted rows!
+     *
+     * @param array $search an assoc array from property to filter value
+     * @param int|null $limit  Output limit
+     * @param int|null $offset Output offset
+     *
+     * @depreacted Legacy function
+     *
+     * @return Entity[]
      */
-    public function findAll(): array
+    public function where(array $search = [], ?int $limit = null, ?int $offset = null)
     {
-        return [];
+        $entity = new $this->entityClass();
+
+        // turn keys into sql query filter, e.g. feedId -> feed_id = :feedId
+        $filter = array_map(
+            function ($property) use ($entity) {
+                // check if the property actually exists on the entity to prevent
+                // accidental Sql injection
+                if (!property_exists($entity, $property)) {
+                    $msg = 'Property ' . $property . ' does not exist on '
+                        . $this->entityClass;
+                    throw new \BadFunctionCallException($msg);
+                }
+
+                $column = $entity->propertyToColumn($property);
+                return $column . ' = :' . $property;
+            },
+            array_keys($search)
+        );
+
+        $andStatement = implode(' AND ', $filter);
+
+        $sql = 'SELECT * FROM `' . $this->getTableName() . '`';
```

Comment:
afaik the ` (backticks) are not suitable for PostgreSQL

Reply:
This was actually copied from the NewsMapper class, if it doesn't work now it never worked to begin with.